app/app.go

```diff
@@ -70,7 +70,11 @@ type App struct {
 }
 
 func (a *App) Run(ctx context.Context) (retErr error) {
-	defer a.Logger.Sync()
+	defer func() {
+		if err := a.Logger.Sync(); err != nil {
+			a.Logger.Error("Unable to flush logs", zap.Error(err))
+		}
+	}()
```

Comment:
There was an unhandled error here, although I'm not sure I've handled this effectively

Reply:
If it cannot flush logs then it probably cannot log :) This is unnecessary.